code/controllers/subsystem/changelog.dm

```diff
@@ -0,0 +1,220 @@
+		to_chat(world, "<span class='alert'>WARNING: Changelog failed to generate. Please inform a coder/server dev</span>")
+		return ..()
+
+	ss_ready = 1
+	// Now we can alert anyone who wanted to check the changelog
+	for(var/client/C in startup_clients_button)
+		UpdatePlayerChangelogButton(C)
+
+	// Now we can alert anyone who wanted to check the changelog
+	for(var/client/C in startup_clients_open)
+		OpenChangelog(C)
+
+	return ..()
+
+	
+/datum/controller/subsystem/changelog/proc/UpdatePlayerChangelogDate(client/C)
+	if(!ss_ready)
+		return // Only return here, we dont have to worry about a queue list because this will be called from ShowChangelog()
+	// Technically this is only for the date but we can also do the UI button at the same time
+	if(preferences_datums[C.ckey].toggles & UI_DARKMODE)
+		winset(C, "rpane.changelog", "background-color=#40628a;font-color=#ffffff;font-style=none")
+	else
+		winset(C, "rpane.changelog", "background-color=none;font-style=none")
+	C.prefs.lastchangelog = current_cl_timestamp
+	var/DBQuery/updatePlayerCLTime = dbcon.NewQuery("UPDATE [format_table_name("player")] SET lastchangelog='[sanitizeSQL(current_cl_timestamp)]' WHERE ckey='[C.ckey]'")
+	if(!updatePlayerCLTime.Execute())
+		var/err = updatePlayerCLTime.ErrorMsg()
+		log_game("SQL ERROR during lastchangelog updating. Error: \[[err]\]\n")
+		message_admins("SQL ERROR during lastchangelog updating. Error: \[[err]\]\n")
+		to_chat(C, "Couldn't update your last seen changelog, please try again later.")
+		return FALSE
+	return TRUE
+
+/datum/controller/subsystem/changelog/proc/UpdatePlayerChangelogButton(client/C)
+	// If SQL aint even enabled, just set the button to default style
+	if(!config.sql_enabled)
+		if(C.prefs.toggles & UI_DARKMODE)
+			winset(C, "rpane.changelog", "background-color=#40628a;text-color=#FFFFFF")
+		else
+			winset(C, "rpane.changelog", "background-color=none;text-color=#000000")
+		return
+
+	// If SQL is enabled but we aint ready, queue them up, and use the default style
+	if(!ss_ready)
+		startup_clients_button += C
+		if(C.prefs.toggles & UI_DARKMODE)
+			winset(C, "rpane.changelog", "background-color=#40628a;text-color=#FFFFFF")
+		else
+			winset(C, "rpane.changelog", "background-color=none;text-color=#000000")
+		return
+
+	// If we are ready, process the button style
+	if(C.prefs.lastchangelog != current_cl_timestamp)
+		winset(C, "rpane.changelog", "background-color=#bb7700;text-color=#FFFFFF;font-style=bold")
+		to_chat(C, "<span class='info'>Changelog has changed since your last visit.</span>")
+	else
+		if(C.prefs.toggles & UI_DARKMODE)
+			winset(C, "rpane.changelog", "background-color=#40628a;text-color=#FFFFFF")
+		else
+			winset(C, "rpane.changelog", "background-color=none;text-color=#000000")
+
+
+/datum/controller/subsystem/changelog/proc/OpenChangelog(client/C)
+	// If SQL is enabled but we aint ready, queue them up
+	if(!ss_ready)
+		startup_clients_open += C
```

Comment:
If you press multiple times you'll be added multiple times to the list. Might want to use |= here